src/Pomelo.EntityFrameworkCore.MySql/Storage/Internal/MySqlTypeMapper.cs

```diff
@@ -60,9 +68,27 @@ public class MySqlTypeMapper : RelationalTypeMapper
 
         readonly Dictionary<string, RelationalTypeMapping> _simpleNameMappings;
         readonly Dictionary<Type, RelationalTypeMapping> _simpleMappings;
-
-        public MySqlTypeMapper()
+        
+        public MySqlTypeMapper(IServiceProvider services)
         {
+            this.services = services;
+            var relationalConn = services.GetService<MySqlRelationalConnection>();
+            var TreatTinyIntAsBoolean = true;
+            var UseDateTime6 = false;
+            if (relationalConn != null)
+            {
+                var csb = new MySqlConnectionStringBuilder(relationalConn.DbConnection.ConnectionString);
+                TreatTinyIntAsBoolean = csb.TreatTinyAsBoolean;
+                relationalConn.DbConnection.Open();
+                using (var cmd = new MySqlCommand("SELECT VERSION();", relationalConn.DbConnection as MySqlConnection))
```

Comment:
relationalConn.DbConnection.ServerVersion should be populated after a connection has been opened [api reference](https://docs.microsoft.com/en-us/dotnet/core/api/system.data.common.dbconnection#System_Data_Common_DbConnection_ServerVersion)

We don't want to require a connection to be opened here in non-scaffolding cases though.  Any way to make the version check depend on scaffolding?